### PR TITLE
Generalise configs to work with yaml files too

### PR DESCRIPTION
**Pull request by jack-willturner**
I personally prefer working with yaml files since I find them a little easier to read/write. 

There are a few other small changes:
- let Rich overflow vertically so we see updated results in the tables 
- add a `-config` shortcut flag to argparse because I'm tired of writing out the whole `filepath_to_json_arguments_config`
- include a couple of small example configs - including an imagenet one which replicates the pytorch default script

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def get_base_argument_parser():
     # logging
     parser.add_argument("--experiment_name", type=str, default="dev")
     parser.add_argument("--logs_path", type=str, default="log")
-    parser.add_argument("--filepath_to_arguments_json_config", type=str, default=None)
+    parser.add_argument("-config", "--filepath_to_arguments_config", type=str, default=None)
```
Comment (AntreasAntoniou) [inline train.py:57]: This is fair, but I have a hunch that some of the code I wrote that takes care of argument parsing does not cover shortcuts. I would be OK with you just changing the default to config. We can do this properly with Hydra, later.

**Review by AntreasAntoniou — CHANGES_REQUESTED** [review]
Assuming you have tested -config in the situations where:
1. You have a config + additional arguments and the additional arguments are supposed to overwrite a couple of arguments that where originally set to defaults (say num gpus to 0).
2. You just have a config without additional arguments.

I would be OK to approve this. Otherwise, just remove the shortcut -config, and rename the longer name to --config.

File: train.py (hunk 1 above)
Comment (jack-willturner) [reply train.py:57]: Ah ok. 

I like the descriptiveness of the longer name, I just don't always want to type it out. I think what I will do is change the long one and add a `help` hint.

Comment (AntreasAntoniou) [reply train.py]: Sounds good. :)

**Review by AntreasAntoniou — APPROVED** [review]

**Pull request merged**